kvbc/src/Replica.cpp

```diff
@@ -539,23 +587,34 @@ std::future<bool> Replica::getBlockAsync(uint64_t blockId, char *outBlock, uint3
 
   auto future = blocksIOWorkersPool_.async(
       [this](uint64_t blockId, char *outBlock, uint32_t *outBlockSize) {
+        bool result = false;
         auto start = std::chrono::steady_clock::now();
         *outBlockSize = 0;
+        LOG_TRACE(logger, "Job Started: " << KVLOG(blockId));
 
         // getBlock will throw exception if block doesn't exist
         try {
-          getBlock(blockId, outBlock, outBlockSize);
+          result = getBlock(blockId, outBlock, outBlockSize);
```

Comment:
isn't it worth to modify the API get block as we discussed to at least have an assertion on the sizes of the in vs out block?

Reply:
There is a dedicated task to change this interface, and get a better idea of the maximal block size + to have the storage layer "knowing" the block size in advance. spoke about it with Gadi as well.
As for the assert, you are right, this is a good time to insert the assert (I wrote it in the other task, but will do it here).
Thanks!

Reply:
@cloudnoize , In the next push I'm going to add another commit adding another argument to getBlock and getBlockAsync to address the above comment.